### ISSUE TITLE
Java: clarify extension naming

### ISSUE THREAD

**Issue by waldyrious** [microsoft/vscode-docs#4786]
The Java Tutorial [currently says](https://github.com/microsoft/vscode-docs/blob/b11c9ea9178999bec683d60635b36437e6b92ac7/docs/java/java-tutorial.md?plain=1#L61) (emphasis mine):

> The supported version for running the **VS Code for Java extension** and the supported version for your projects are two separate runtimes.

However, there doesn't seem to be a "VS Code for Java extension". I raised this [in #4754](https://github.com/microsoft/vscode-docs/pull/4754#discussion_r688455202) (and [in #4003](https://github.com/microsoft/vscode-docs/pull/4003#discussion_r688440618), which is where this language was first introduced), but it warrants a separate issue for proper discussion.

At first I thought this could have been an old name for the [Java extension pack](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack), but on a closer look, it appears to be specifically related to the Java language server, which IIUC is provided by the [Language Support for Java](https://marketplace.visualstudio.com/items?itemName=redhat.java) extension (which is included in the extension pack). Indeed, that page says exactly this:

> (latest Java 11 is the current minimum requirement).

This is consistent with what's written in [java-project.md](https://github.com/microsoft/vscode-docs/blob/main/docs/java/java-project.md):

> Although the Java language server requires JDK version 11 or above to run, this is NOT a requirement to your project's runtime.

It also matches what the configuration wizard says:

![Screenshot from 2021-08-13 18-21-59_](https://user-images.githubusercontent.com/478237/129397341-7bdae0d5-2625-4dfc-aa9e-348e241097ac.png)

(Although clearly the second highlighted message is too ambiguous and should be fixed too — I've submitted https://github.com/microsoft/vscode-java-pack/issues/729 to track that.)

**TL;DR**: It seems that we should change the expression "VS Code for Java extension" in the [Java Tutorial](https://github.com/microsoft/vscode-docs/blob/b11c9ea9178999bec683d60635b36437e6b92ac7/docs/java/java-tutorial.md#supported-java-versions) page to explicitly reference the `redhat.java` extension. As to what phrasing/naming would be appropriate ("The `redhat.java` extension"? "the Language Support for Java extension"? etc.), I suppose it might make sense to wait for the resolution of https://github.com/microsoft/vscode-java-pack/issues/729 and use the same wording, to be consistent.

**Comment by gregvanl** [comment]
@nickzhums for visibility
